lib/sentry/event.ex

```diff
@@ -208,18 +218,14 @@ defmodule Sentry.Event do
     |> Enum.reverse()
   end
 
-  @spec put_source_context(map(), String.t(), integer()) :: map()
-  def put_source_context(frame, file, line_number) do
-    if @source_code_context_enabled do
-      {pre_context, context, post_context} =
-        Sentry.Sources.get_source_context(@source_files, file, line_number)
+  @spec macro_put_source_context(map(), String.t(), integer()) :: map()
+  def macro_put_source_context(frame, file, line_number) do
```

Comment:
make this private and rename it to "do_put_source_context" instead of macro. Its pretty common.